KPI-team1-frontend/src/pages/SettingsPage.tsx

```diff
@@ -0,0 +1,154 @@
+        p_default_circle_id: newDetails.defaultCircleId || null
+      })
+
+    setUserDetails(newDetails);
+    navigate(newDetails?.defaultCircleId ? "/kpi/" + newDetails.defaultCircleId : "/");
+  }
+
+  function resetAndNavigateBack() {
+    setUsername(userDetails.username)
+    setDefaultCircleId(userDetails.defaultCircleId)
+
+    navigate(-1)
+  }
+
+  return (
+    <form onSubmit={(e: React.SyntheticEvent) => handleSubmit(e)}>
+
+      <div className="bg-grey-800 p-4 mb-4 border-b border-gray-300">
+        <div className="flex justify-between items-center">
+          <h1 className="text-3xl">Settings</h1>
+          <div>
+            <button
+              onClick={() => resetAndNavigateBack()}
+              className="w-28 h-10 mr-4 bg-white rounded border border-customYellow justify-center items-center gap-2 inline-flex text-zinc-700 text-base font-medium"
+            >
+              Cancel
+            </button>
+            <button
+              className="w-28 h-10 bg-customYellow rounded justify-center items-center gap-2 inline-flex text-base font-medium"
+            >
+              Save
+            </button>
+          </div>
+        </div>
+      </div>
+
+      <div className="flex">
+        <div className="w-1/2 bg-white shadow-sm">
+          <div className="p-2 bg-customPurple rounded-t">
+            <h2 className="text-[#131313] font-bold m-1">Username and circles</h2>
+          </div>
+          <div className="p-4 rounded-b">
+            <div className="flex flex-col text-sm mb-5">
+              <label className="font-bold mb-2 text-gray-800" htmlFor="username">
+                Username
+              </label>
+              <input
+                name="username"
+                type="text"
+                placeholder="Username"
+                value={username ? username : ''}
+                onChange={(e) => setUsername(e.target.value)}
+                className="bg-[#FFF] shadow-sm border border-gray-200 p-2 focus:outline-none focus:border-gray-500 rounded-lg"
+              />
+            </div>
+            <div className="flex flex-col text-sm mb-5">
+              <label className="font-bold mb-2 text-gray-800" htmlFor="defaultCircle">
+                Default circle
+              </label>
+              {circles.length > 0 ? (
+                <select
+                  name="defaultCircle"
+                  defaultValue={defaultCircleId ? defaultCircleId : ''}
+                  onChange={(e) => {
+                    setDefaultCircleId(e.target.value);
+                  }}
+                  className="bg-[#FFF] border border-gray-300 text-gray-900 text-sm rounded-lg focus:outline-none focus:border-gray-500 block w-full p-2"
+                >
+                  <option value=""></option>
+                  {circles.map((c) => (
+                    <option
+                      value={c.circle_user[0].circle_id}
+                    // TODO provide random key
```

Comment:
add random key:
` {circles.map((c, index) => (
                    <option
                      value={c.circle_user[0].circle_id}
                      key={index}
                    >
                      {c.circle_name}
                    </option>`

Reply:
But that is not random unfortunately :(